src/Components/Notes/Notes.jsx

```diff
@@ -0,0 +1,84 @@
+import React, { useEffect, useState } from 'react';
+import SearchBar from '../SearchBar/SearchBar';
+import notesData from '../data/data';
+import NoteItem from '../NoteItem/NoteItem';
+import AddNote from '../AddNote/AddNote';
+import { format } from 'date-fns';
+import './notes.css'
+function Notes() {
+  const [notes, setNotes] = useState(notesData);
+  const [filterNotes, setfilterNotes] = useState(notes);
```

Comment:
I prefer to have a derived state from original notes based on the search value because you need to keep managing the two states (original items and filtered items) and this can introduce a complexity, especially when updating or syncing them.
So when adding/updating/filtering, you need to deal only with `originalNotes` and `searchValue`, and the rendered `notes` will compute each time.

and let's simplify the filter logic by using Array instead.

So I can suggest the following:

```suggestion
  const [originalNotes, setOriginalNotes] = useState(notesData);
  const [searchValue, setSearchValue] = useState(""); // State for storing search, filters, ...
  const notes = React.useMemo(() =>
    originalNotes.filter(
      (note) =>
        [note.title, note.content, note.creation_date].some((data) =>
          data.toLocaleLowerCase().includes(searchValue)
        ),
      [originalNotes, searchValue]
    )
  );
```


By applying this suggestion, you need to pass the `searchValue` to `SearchBar` component instead of the local state.